File: Portfolio/Portfolio.Blazor/Data/Entities/ResumeTranslation.cs
using Portfolio.Blazor.Data;

namespace Portfolio.Blazor.Data.Entities;

public sealed class ResumeTranslation
{
    public int Id { get; set; }
    public int ResumeId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string? Summary { get; set; }

    [JsonOrEmpty]
    public string? Leadership { get; set; }

    [JsonOrEmpty]
    public string? Education { get; set; }

    [JsonOrEmpty]
    public string? Certificates { get; set; }

    [JsonOrEmpty]
    public string? Certifications { get; set; }

    [JsonOrEmpty]
    public string? Publications { get; set; }

    [JsonOrEmpty]
    public string? Recommendations { get; set; }

    [JsonOrEmpty]
    public string? TechnicalProductions { get; set; }

    [JsonOrEmpty]
    public string? Events { get; set; }

    [JsonOrEmpty]
    public string? Awards { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Resume? Resume { get; set; }
}

File: src/laravel/app/Models/ResumeTranslation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ResumeTranslation extends Model
{
    use HasFactory;

    protected $fillable = ['resume_id', 'locale', 'summary', 'leadership', 'education', 'certificates', 'certifications', 'publications', 'recommendations', 'technical_productions', 'events', 'awards'];

    protected $casts = ['leadership' => 'array', 'education' => 'array', 'certificates' => 'array', 'certifications' => 'array', 'publications' => 'array', 'recommendations' => 'array', 'technical_productions' => 'array', 'events' => 'array', 'awards' => 'array'];

    /**
     * @return BelongsTo<Resume, $this>
     */
    public function resume(): BelongsTo
    {
        return $this->belongsTo(Resume::class);
    }
}
